zlib-rs/src/deflate/algorithm/fast.rs

```diff
@@ -33,59 +29,66 @@ pub fn deflate_fast(stream: &mut DeflateStream, flush: DeflateFlush) -> BlockSta
         // Insert the string window[strstart .. strstart+2] in the
         // dictionary, and set hash_head to the head of the hash chain:
 
+        let lc: u8; // Literal character to output if there is no match.
         if state.lookahead >= WANT_MIN_MATCH {
-            let hash_head = StandardHashCalc::quick_insert_string(state, state.strstart);
-            dist = state.strstart as isize - hash_head as isize;
+            let val = u32::from_le_bytes(
+                state.window.filled()[state.strstart..state.strstart + 4]
+                    .try_into()
+                    .unwrap(),
+            );
+            let hash_head = StandardHashCalc::quick_insert_value(state, state.strstart, val);
+            let dist = state.strstart as isize - hash_head as isize;
 
-            /* Find the longest match, discarding those <= prev_length.
-             * At this point we have always match length < WANT_MIN_MATCH
-             */
+            // Find the longest match for the string starting at offset state.strstart.
```

Comment:
did we lose the property that

> At this point we have always match length < WANT_MIN_MATCH

I'd assume not, in which case I'd like to keep that bit of the comment.